### ISSUE TITLE
Purple Box Issue

### ISSUE THREAD

**Issue by DanielTheDev** [logisim-evolution/logisim-evolution#1706]
Every time I load my project, the purple boxes around each element appear and I cannot get rid of them unless I reset the simulation.
No matter what I do, they keep appearing when I load my project. Besides, if I move them they appear glitched.
How can I fix the purple boxes?

![image](https://user-images.githubusercontent.com/36048138/229806202-69f15448-6011-4b2c-b315-fa932d1e8d50.png)

**Comment by juli-99** [comment]
Could you please provide the `.circ` and more information?
What platform are you on (Win, Mac, Linux)?
How did you install LSe?
What Java version do you use (`java --version`)?
Did you try it with the current head of develop?

**Comment by DanielTheDev** [comment]
> Could you please provide the `.circ` and more information? What platform are you on (Win, Mac, Linux)? How did you install LSe? What Java version do you use (`java --version`)? Did you try it with the current head of develop?

No, I did not try that version. Additionally, the error only occurred after a couple of days. So not in the beginning.

**OS**: Windows 11
**Install**: logisim-evolution-3.8.0-all.jar
**Version**: 3.8.0

[CPU.circ](https://github.com/logisim-evolution/logisim-evolution/files/11152007/CPU.txt)

**Comment by maehne** [comment]
@DanielTheDev: I am not able to reproduce this issue in Logisim-evolution v3.8.0dev (ID:develop/3e1efb0c) on Windows 11. Could you please try the [released MSI installer](https://github.com/logisim-evolution/logisim-evolution/releases/tag/v3.8.0)? Could you please check whether changing the Graphics acceleration setting in Window -> Preferences -> Experimental has an influence?

**Comment by maehne** [comment]
Thanks for the additonal details. It has indeed something to do with the auto-propagate functionality. Apparently, the purple boxes signal activation of the corresponding block and LSe doesn't keep properly track of them once they are moved.

**Comment by DanielTheDev** [comment]
> Thanks for the additonal details. It has indeed something to do with the auto-propagate functionality. Apparently, the purple boxes signal activation of the corresponding block and LSe doesn't keep properly track of them once they are moved.

I do have another question. 
Does the auto-propagate setting stays withing the java process or should it be stored in the .circ file or the java preferences? Because I don't really like to have to turn it off everytime I start the program.

**Comment by maehne** [comment]
> Does the auto-propagate setting stays within the java process or should it be stored in the .circ file or the java preferences? Because I don't really like to have to turn it off everytime I start the program.

As far as I can tell, this setting is project-specific and not stored as a property into the `.circ` file. I have mixed feelings about adding this feature. On the one hand, I can absolutely see the point, why one would have it saved on the other hand, it would cause an additional reason, which causes a `.circ` file to become outdated and requiring saving. IMHO, changing simulation related settings should leave a `.circ` file untouched.
